test/models/attendance_test.exs

```diff
@@ -5,20 +5,48 @@ defmodule CoursePlanner.AttendanceTest do
   import CoursePlanner.Factory
 
   @valid_class_attrs %{offered_course: nil, date: %{day: 17, month: 4, year: 2010}, finishes_at: %{hour: 14, min: 0, sec: 0}, starting_at: %{hour: 15, min: 0, sec: 0}}
-  @valid_attrs %{student_id: nil, class_id: nil, attendance_type: "some content"}
+  @valid_attrs %{student_id: nil, class_id: nil, attendance_type: "some content", comment: ""}
   @invalid_attrs %{}
 
-  test "changeset with valid attributes" do
+  setup do
+    insert(:system_variable, key: "ATTENDANCE_DESCRIPTORS", value: "sick_leave, informed_beforehand", type: "list")
+
     student = insert(:user, %{role: "Student"})
     offered_course = insert(:offered_course, %{students: [student]})
     class = insert(:class, %{@valid_class_attrs | offered_course: offered_course})
 
-    changeset = Attendance.changeset(%Attendance{}, %{@valid_attrs | student_id: student.id, class_id: class.id})
+    {:ok, %{student: student, class: class}}
+  end
+
+  test "changeset with valid attributes", context do
+    changeset = Attendance.changeset(%Attendance{}, %{@valid_attrs | student_id: context.student.id, class_id: context.class.id})
     assert changeset.valid?
   end
 
-  test "changeset with invalid attributes" do
+  test "changeset with invalid attributes", _context do
     changeset = Attendance.changeset(%Attendance{}, @invalid_attrs)
     refute changeset.valid?
   end
+
+  describe "attendance comment" do
+    test "passes when comment is empty", context do
```

Comment:
It's better to pattern match the context like `%{student: student, class: class}` because if you change the `setup` the error is clearer.